cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -339,6 +374,95 @@ impl YaccParser {
         }
     }
 
+    fn parse_yacckind(&mut self, i: usize, update_yacc_kind: bool) -> Result<usize, YaccGrammarError> {
+        const YACC_KINDS: [(&str, YaccKind); 5] = [
+            ("Grmtools", YaccKind::Grmtools),
+            (
+                "Original(NoAction)",
+                YaccKind::Original(YaccOriginalActionKind::NoAction),
+            ),
+            (
+                "Original(UserAction)",
+                YaccKind::Original(YaccOriginalActionKind::UserAction),
+            ),
+            (
+                "Original(GenericParseTree)",
+                YaccKind::Original(YaccOriginalActionKind::GenericParseTree),
+            ),
+            ("Eco", YaccKind::Eco),
+        ];
+        let j = self.parse_ws(i, false)?;
+        let s = &self.src[i..];
+        for (kind_name, kind) in YACC_KINDS {
+            if s.starts_with(kind_name) {
+                if update_yacc_kind {
+                    self.yacc_kind = kind;
+                }
+                let end_pos = j + kind_name.len();
+                return Ok(end_pos);
+            }
+        }
+        // This strikes me as an interesting case where we know the start pos of the
```

Comment:
Agreed. In general, I tend to think that the user is helped most by pointing out the earliest possible place that an error is detected, so we're doing the right thing here I think.